src/main/autoStart.ts

```diff
@@ -10,38 +10,51 @@ import { join } from "path";
 
 interface AutoStart {
     isEnabled(): boolean;
+    wasAutoStarted(): boolean;
     enable(): void;
     disable(): void;
 }
 
+const isFlatpak = process.env.FLATPAK_ID !== undefined;
+
 function makeAutoStartLinux(): AutoStart {
     const configDir = process.env.XDG_CONFIG_HOME || join(process.env.HOME!, ".config");
     const dir = join(configDir, "autostart");
     const file = join(dir, "vencord.desktop");
 
     return {
-        isEnabled: () => existsSync(file),
+        isEnabled: () => existsSync(file), // TODO: flatpak
+        wasAutoStarted: () => process.argv.includes("--autostart"),
         enable() {
-            const desktopFile = `
+            if (isFlatpak) {
```

Comment:
not sure if this logic is good. why not unconditionally first try using the portal and if it doesn't work, use the desktop fallback instead? lots of DEs implement the portal so it should work on those even without flatpak, no?

Reply:
i havent tested gdbus yet fully so this is just placeholder code